paimon-core/src/main/java/org/apache/paimon/utils/BranchManager.java

```diff
@@ -142,6 +145,74 @@ public boolean fileExists(Path path) {
         }
     }
 
```

Comment:
Do we need to change the `latest` snapshot hint? You can see the code of `RollbackHelper::cleanLargerThan`. And do we need to delete the datafiles and manifests related to the to-be-deleted snapshots? If not, is there any other way to delete the orphan files?

Reply:
> Do we need to change the `latest` snapshot hint? You can see the code of `RollbackHelper::cleanLargerThan`. And do we need to delete the datafiles and manifests related to the to-be-deleted snapshots? If not, is there any other way to delete the orphan files?

@schnappi17 Use "CALL sys. remove_orphan_files" to delete orphan files?

Reply:
I previously created a pr about deleting orphan files：https://github.com/apache/incubator-paimon/pull/2863